x-pack/plugin/ingest/src/test/java/org/elasticsearch/xpack/ingest/FingerprintProcessorTests.java

```diff
@@ -0,0 +1,438 @@
+    }
+
+    public void testMapEntriesAreConsistentlyOrdered() throws Exception {
+        List<String> keyList = randomList(1, 10, () -> randomAlphaOfLength(8));
+        List<String> sortedKeyList = new ArrayList<>(keyList);
+        sortedKeyList.sort(Comparator.naturalOrder());
+
+        var sortedInputMap = new LinkedHashMap<String, Object>();
+        List<Object> expectedValues = new ArrayList<>();
+        for (String s : sortedKeyList) {
+            sortedInputMap.put(s, s);
+            expectedValues.add(s);
+            expectedValues.add(s);
+        }
+        var docMap = new HashMap<String, Object>();
+        docMap.put("map", sortedInputMap);
+        String sortedFingerprint = doTestFingerprint(docMap, List.of("map"), expectedValues, null);
+
+        var shuffledInputMap = new LinkedHashMap<String, Object>();
+        for (String s : keyList) {
+            shuffledInputMap.put(s, s);
+        }
+        docMap = new HashMap<String, Object>();
+        docMap.put("map", shuffledInputMap);
+        String shuffledFingerprint = doTestFingerprint(docMap, List.of("map"), expectedValues, null);
+
+        assertThat(sortedFingerprint, equalTo(shuffledFingerprint));
+    }
+
+    public void testIgnoreMissing() throws Exception {
+        // only one value contributes to fingerprint
+        var docMap = new HashMap<String, Object>();
+        docMap.put("foo", "foo");
+        doTestFingerprint(docMap, List.of("foo", "bar", "baz"), List.of("foo"), "WoyqQDn9vALAGmScjA9Z2yg7sos=", true);
+
+        // two values contribute to fingerprint
+        docMap = new HashMap<>();
+        docMap.put("foo", "foo");
+        docMap.put("bar", "foo");
+        doTestFingerprint(docMap, List.of("foo", "bar", "baz"), List.of("foo", "foo"), "vjq2RyU5UA8vzeM5gIbfrOGir7w=", true);
+
+        // three values contribute to fingerprint
+        docMap = new HashMap<>();
+        docMap.put("foo", "foo");
+        docMap.put("bar", "foo");
+        docMap.put("baz", "foo");
+        doTestFingerprint(docMap, List.of("foo", "bar", "baz"), List.of("foo", "foo", "foo"), "2Ozd89kaee2AnbrjU8zB6QGn9Wo=", true);
+
+        // error when ignore_missing is false
+        final var docMap2 = new HashMap<String, Object>();
+        docMap2.put("foo", "foo");
+        docMap2.put("bar", "foo");
+        IllegalArgumentException e = expectThrows(
+            IllegalArgumentException.class,
+            () -> doTestFingerprint(docMap2, List.of("foo", "bar", "baz"), List.of("foo"), null, false, null)
+        );
+        assertThat(e.getMessage(), containsString("missing field [baz] when calculating fingerprint"));
+    }
+
+    public void testDataTypes() throws Exception {
+        var typesMap = new HashMap<String, Object>();
+        typesMap.put("0string", "foo");
+        typesMap.put("1byte[]", new byte[] { 0, 1, 2 });
+        typesMap.put("2integer", 42);
+        typesMap.put("3long", 43L);
+        typesMap.put("4float", 3.14F);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [CNT: Rough value of known constant found (CNT_ROUGH_CONSTANT_VALUE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#cnt-rough-value-of-known-constant-found-cnt-rough-constant-value).
It's recommended to use the predefined library constant for code clarity and better precision.